app/models/tag.rb

```diff
@@ -86,11 +86,13 @@ def self.parse(list)
     end
   end
 
+  # @option options :ns [String, nil]
+  # @option options :cat [String, nil] optional category to add to the end (with a slash)
+  # @return [String] downcases namespace or category
   def self.get_namespace(options)
-    options = {:ns => '/user'}.merge(options)
-    ns = options[:ns]
-    ns = "" if [:none, "none", "*", nil].include?(options[:ns])
-    options[:cat].nil? ? ns.downcase : [ns, options[:cat]].join("/").downcase
+    ns = options.key?(:ns) ? options[:ns] : '/user'
+    ns = "" if [:none, "none", "*", nil].include?(ns)
+    options[:cat] ? "#{ns}/#{options[:cat]}".downcase : ns.downcase
```

Comment:
```ruby
ns = options.fetch(:ns, '/user')
ns = "" if [:none, "none", "*", nil].include?(ns)
ns += "/" + options[:cat] if options[:cat]
ns.downcase
```

I don't feel strongly about the cat/downcase split, though replacing the ternary with a second (more symmetrical?) trailing conditional seems a little nicer to me.

`fetch` beats `key?`+`[]` every time.